assets/src/domain/shared/services/utils/switchTenseForDate/index.ts

```diff
@@ -0,0 +1,44 @@
+import { isNil } from 'ramda';
+import { parseISO } from 'date-fns';
+
+import { diff } from '@appServices/utilities';
+import { now } from '@sharedServices/utils/dateAndTime';
+
+interface CalendarDateLabels {
+	headerPast: string;
+	headerFuture: string;
+	footerPast: string;
+	footerFuture: string;
+}
+
+interface SwitchTenseForDate extends CalendarDateLabels {
+	endDate: string; // ISO string
+	startDate: string; // ISO string
+}
+
+const switchTenseForDate = ({
+	endDate,
+	footerFuture,
+	footerPast,
+	headerFuture,
+	headerPast,
+	startDate,
+}: SwitchTenseForDate): { footerText: string; headerText: string } => {
```

Comment:
Ya my apologies because I didn't explain this well at all either...

From that comment in 2614 it was totally clear to me what

```js
const headerText = switchTenseForDate( date, labels.headerPast, labels.headerFuture );
```

meant, but in retrospect I can see that it's not clear at all.

So in my mind `switchTenseForDate()` would ge a generic utility function within the application layer, meaning it would not be specific to any particular use case (domain).
I saw this function being something like:

```js
const switchTenseForDate  = ( date, textForPastDate, textForFutureDate ) => {
	// using the logic you created
	return diff('minutes', parseISO(date), now) < 0 ? textForPastDate: textForFutureDate;
}
```

this could then be used anywhere for any date and for any purpose and would not be specific to the Editor or the `CalendarDateSwitcher` component.

For our current requirements in the `CalendarDateSwitcher` component, we would have to call this utility function for both the start date and end date in order to decide whether to use `footerFuture` or `footerPast` and `headerFuture` or `headerPast` which would be passed into the component via the `CalendarDateLabels` object.

Again, sorry for being "clear as mud" with my previous comments